api/app/signals/apps/api/serializers/signal.py

```diff
@@ -542,6 +543,30 @@ def get__display(self, obj):
         return obj.sia_id
 
 
+class PublicSignalGeographtSerializer(serializers.Serializer):
+    """"
+    Custom serializer to validate if either bbox or lat/lon is set
+    """
+
+    bbox = serializers.CharField(required=False)
+    lat = serializers.FloatField(required=False)
+    lon = serializers.FloatField(required=False)
+
+    maincategory_slug = serializers.CharField()
+    category_slug = serializers.CharField()
+
+    def validate(self, attrs: Dict) -> Dict:
```

Comment:
This function either returns a dict or raises an error, so the typing should be `) -> Union[Dict, NoReturn]:`. Right?